core/src/main/java/org/bitcoinj/evolution/SubTxTransition.java

```diff
@@ -42,6 +58,35 @@ protected void parse() throws ProtocolException {
         length = cursor - offset;
     }
 
+    @Override
```

Comment:
I don't see this method as being necessary and it doesn't follow the conventions of Message.unsafeBitcoinSerialize because it serializes data that is not included in this object.

Can you instead use Transaction.unsafeBitcoinSerialize with a Transaction object that has the extra payload data of type SubTxTransition.

```
Transaction tx = new Transaction(params);
tx.setExtraPayload(new SubTxTransition(...));
byte [] bytes = tx.unsafeBitcoinSerialize();
```

What is the reason for having this function?  What are you doing with the data?